lib/synapse_pay_rest/http_client.rb

```diff
@@ -85,44 +78,35 @@ def update_headers(oauth_key: nil, fingerprint: nil, client_id: nil,
       nil
     end
 
-    def ssl_certicate
-      pem = @development_mode ? @vgs_sandbox_pem : @vgs_live_pem
-      OpenSSL::X509::Certificate.new(pem)
-    end
+    def request_params(method, url:, payload: nil, idempotency_key: nil, tunnel: false)
+      params = { method: method, url: url, headers: headers }
 
-    def tunnel_params
-      {
-        ssl_client_cert: ssl_certicate,
-        verify_ssl: OpenSSL::SSL::VERIFY_NONE,
-        proxy: @vgs_url
-      }
-    end
+      params[:payload] = payload.to_json if payload.present?
+      params[:headers].merge!('X-SP-IDEMPOTENCY-KEY' => idempotency_key) if idempotency_key.present?
 
-    def request_params(method, url:, payload: nil, tunnel: false, idempotency_key: nil)
-      request_headers = headers
-      if idempotency_key
-        request_headers.merge!({'X-SP-IDEMPOTENCY-KEY' => idempotency_key})
-      end
+      return vgs_proxy_request_params(base_params: params) if tunnel
+      params
+    end
 
-      params = {
-        method: method,
-        url: url,
-        headers: request_headers
-      }
+    def vgs_proxy_request_params(base_params:)
+      base_params.merge(
+        ssl_client_cert: OpenSSL::X509::Certificate.new(vgs_pem),
+        verify_ssl: OpenSSL::SSL::VERIFY_NONE,
+        proxy: vgs_url,
 
-      params[:payload] = payload.to_json if payload.present?
-      params.merge!(tunnel_params) if tunnel
-      params
+        # NOTE: next line is just defensive programming, making sure the client_id and client_secret are not sent to VGS.
```

Comment:
this is great, but can i suggest a slight language change: remove the "just" and make the wording more forceful, like "under no circumstances do we want to send the client_id and client_secret to VGS"